backend/src/dao/TeamEventAttendanceDao.ts

```diff
@@ -58,26 +59,12 @@ export default class TeamEventAttendanceDao {
    */
   static async deleteAllTeamEventAttendance(): Promise<void> {
     const snapshot = await teamEventAttendanceCollection.get();
-    await this.deleteAllTeamEventAttendanceHelper();
+    await deleteCollection(db, 'team-event-attendance', 500);
     if (snapshot.size !== 0) {
       await this.deleteAllTeamEventAttendance();
     }
```

Comment:
I don't think this is needed anymore

Reply:
+1. The logic in `deleteCollection` already takes care of ensuring that all the documents are deleted. You specify the batch size for the number in each batch that we delete (500 is the limit), and the function will keep deleting in batches of size 500. You don't need to recurse further.